Update Progress Bar for Stencil 2.0 and Adaptive 2.1

cdegit opened the pull request:
> Progress Bar is a horizontal indicator used to visualize the progression of an extended operation, such as a download, file transfer, or installation. It may be accompanied by a textual representation of the progress in a percent format. When a process may take a while but where values are indeterminate, use a Loader or Spinner instead. For typical progress UI at the top of checkouts, use the Progress Steps component instead.

Status: Ready for review

Reviewers: @jeffkamo @fractaltheory 
## Jira Tickets
- [x] https://mobify.atlassian.net/browse/STEN-21
## To Dos
- [ ] +1
- [x] Get animation for radial progress bar working
- [x] SASS vars to be overridden
- [x] Add support for RTL
- [ ] Replace font awesome with the icon component
## Feedback
- [x] Fix instructions
- [x] Fix SASS lint errors
- [x] Make structure better match our standards
- [x] Modifier class to show icons vs text
## How to Test
- Clone the repo
- Run `sudo npm install` and `bower install`
- Make a custom build of zepto: 

```
cd bower_components/zeptojs
npm install
npm run-script dist
MODULES="zepto event data touch" npm run-script dist
```
- Run `grunt serve`
- Visit http://localhost:3000/tests/visual/

Commit:
Initial commit

# Conflicts:
#	README.md

Commit:
Initial markup, styling and behaviour

Commit:
Added some basic visual tests

Commit:
Allow an individual element to be passed to ui functions

Commit:
Add state and input tests

Commit:
Fix issue where spinners would use wrong clippath

Commit:
Cleaned up css

Commit:
Add aria role and attributes

Commit:
Update schema

Commit:
Switch from shoppicon to an icon font

Needed to be able to set the icon colors in a way that would work for older android devices.

Commit:
Added retry button

Commit:
Added -webkit prefix to setProgress

Commit:
Improved text positioning inside spinner

Commit:
Expose script via the DOM

Commit:
Split into subcomponent templates

Commit:
Add readme scaffold

Commit:
Add part of API to the readme

Commit:
Remove publicly accessible init, fix bindEvents

Commit:
Add template param info to readme

Commit:
Improve zepto instructions in readme

Commit:
Fix adaptivejs dependency

Commit:
Fix instructions in readme

cole-sanderson commented on README.md:
````diff
@@ -1 +1,137 @@
+* Run `npm install && bower install && bundle install`
+* Compile a custom build of zepto. Run the following:
+``` 
+cd bower_components/zeptojs
````
change to zeptojs

jeffkamo replied:
what do you mean?

cole-sanderson replied:
the path was `zepto` and now its changed to `zeptojs`

jeffkamo replied:
isn't that what it already says?

cole-sanderson replied:
if you look at this commit: https://github.com/mobify/stencil-progress-bar/commit/a14757cbf84e8ad20b4b98a325ea1a25246a8c8d and you will see the change.

jeffkamo replied:
Okay, so I'm not clear why you're pointing it out if it's already done? Maybe I am misunderstanding the intent of your comment there?

cole-sanderson replied:
Yeah, I am not sure either. I got confused at first place, not too sure how that happened.

cole-sanderson commented on README.md:
```diff
@@ -1 +1,137 @@
+
+## Development
+
+* Run `npm install && bower install && bundle install`
```
add sudo to avoid some issues

jeffkamo replied:
I believe `sudo` has it's own issues that can occur. I think it's fine to say just `npm install` etc. Users can choose to `sudo` the command at their own risk.

ry5n replied:
I agree with Jeff, we shouldn’t encourage sudo.

contextlimit replied:
In my experience, I don't need `sudo`. When I've seen people using it, it was because their environment permissions were kind of mixed up.

cole-sanderson commented on README.md:
```diff
@@ -1 +1,137 @@
+
+## Usage
+
+Stencil components installed with the AdaptiveJS generator require no setup. Just load the component’s template in the desired view and use the dust helper to render it. For details, see [using Components in AdaptiveJS](#).
```
Add link here?

jeffkamo replied:
Documentation is still in beta. Not sure if we should link to the current beta docs (not sure if the URLs will change).

cole-sanderson commented on README.md:
```diff
@@ -1 +1,137 @@
+
+A progress bar that comes in both standard and radial flavours. 
+
+[Link to demo](#)
```
Add link here?

jeffkamo replied:
We haven't yet determined how demos are going to be hosted, so this feature will have to wait.

cole-sanderson commented on README.md:
```diff
@@ -1 +1,137 @@
+
+## Tests
+
+### Visual Tests
```
Getting blank page when I follow this steps

ry5n replied:
This worked for me, but I didn’t need to run the first `npm run-script dist`. I did the following after a clean clone of this repo:

``` shell
cd stencil-progress-bar
git checkout create-progress-bar
npm install && bower install && bundle install
cd bower_components/zeptojs
npm install
MODULES="zepto event data touch" npm run-script dist
cd ../../
grunt serve
```

cole-sanderson commented on _progress-bar.scss:
```diff
@@ -0,0 +1,206 @@
+$progress-bar__variable-name: true !default;
+
+// Override Font Awesome's default path
+$fa-font-path: "/bower_components/font-awesome/fonts";
```
scss-lint issue: change to single quote

cole-sanderson commented on _progress-bar.scss:
```diff
@@ -0,0 +1,206 @@
+    height: $tap-size;
+    margin: 0 15px;
+
+    align-items: center;
```
scss-lint issue: move align-items: center after display

cole-sanderson commented on _progress-bar.scss:
```diff
@@ -0,0 +1,206 @@
+        position: absolute;
+        top: 50%;
+        left: 50%;
+        transform: translate(-50%, -50%);
```
scss-lint issue: add space to separate from position group

cole-sanderson commented on _progress-bar.scss:
```diff
@@ -0,0 +1,206 @@
+.c-progress-bar__bar-section {
+    margin-right: 5px;
+
+    flex-grow: 1;
```
scss-lint issue: move flex-grow to top

cole-sanderson commented on _progress-bar.scss:
```diff
@@ -0,0 +1,206 @@
+    }
+
+    .c-progress-bar__progress {
+        transform: translate3d(-100%, 0, 0);
```
scss-lint issue: move transform before transition

cole-sanderson commented on _progress-bar.scss:
```diff
@@ -0,0 +1,206 @@
+        position: absolute;
+        top: 50%;
+        left: 50%;
+        transform: translate(-50%, -50%);
```
scss-lint issue: add space to separate from position group

jeffkamo replied:
@cole-sanderson for future reference, try to avoid linting warnings or errors in PR comments. While important, they pull focus away from criticizing the actual code itself.

cole-sanderson commented on _progress-bar.scss:
```diff
@@ -0,0 +1,206 @@
+}
+
+
+// Visually Hidden
```
Visually hidden should not live in progress-bar.scss. Will include stencil utilities to include visually hidden style.

Commit:
Get basic radial animation working

Commit:
Added stencil utilites for visually hidden style

Commit:
Merge branch 'create-progress-bar' of github.com:mobify/stencil-progress-bar into create-progress-bar

cole-sanderson commented on _progress-bar.scss:
```diff
@@ -0,0 +1,188 @@
+@import 'stencil-utils/dist/visibility/visibility';
+
+
+@mixin state-colors($color) {
```
I am not sure if we should include mixin in progress-bar. I think instead of mixin, add modifier class (c--success, c--error, etc) to class.

cole-sanderson commented on _progress-bar.scss:
```diff
@@ -0,0 +1,188 @@
+}
+
+
+.c-progress-bar--state-inprogress {
```
Change modifier class to 'c--progress', 'c--success', and 'c--error' and add style in class

cole-sanderson commented on progress-bar.dust:
```diff
@@ -0,0 +1,30 @@
+<div 
+    {?id}id="{.}"{/id} 
+    class="c-progress-bar 
+        c-progress-bar--state-{state} 
```
change c-progress-bar--state to just {state} = 'c--error', 'c--success', etc.

cole-sanderson commented on progress-bar.dust:
```diff
@@ -0,0 +1,30 @@
+    {?id}id="{.}"{/id} 
+    class="c-progress-bar 
+        c-progress-bar--state-{state} 
+        {?useSpinner}c-progress-bar--spinner-enabled{/useSpinner} 
```
change c-progress-bar--spinner-enabled to c--spin

Commit:
Add counterclockwise animation

Commit:
Remove comment from bower.json

Remember to update stencils-util to the master branch when it is ready

cole-sanderson commented on _progress-bar.scss:
```diff
@@ -0,0 +1,188 @@
+
+}
+
+.c-progress-bar--spinner-enabled {
```
change to' c--spin' modifier class add modifier style in c-progress-bar class

Commit:
Update readme following feedback

cole-sanderson commented on progress-bar.dust:
```diff
@@ -0,0 +1,30 @@
+        {@c-progress-bar__spinner /}
+    {/useSpinner}
+
+    <div class="c-progress-bar__bar-section">
```
suggestion change for this block:

`<div class="c-progress-bar__bar">
            <div class="c-progress-bar__label">{label}</div>
            <div class="c-progress-bar__block">
                <div class="c-progress-bar__progress">
                    <span class="u-visually-hidden">{progress-text}</span> {! add text here for screen reader !}
                </div>
            </div>
        </div>`

Commit:
Refactor scss and rename states

cole-sanderson commented on progress-bar.dust:
```diff
@@ -0,0 +1,30 @@
+        </div>
+    </div>
+
+    <div class="c-progress-bar__extra-section">
```
suggestion change for this block:

`<div class="c-progres-bar__extra {modifier}">
            <div class="c-progress-bar__text"></div>
            <i class="c-progress-bar__icon c--check"></i>
            <button class="c-progress-bar__icon c--retry js-retry"></button>
        </div>`

cole-sanderson replied:
Add style for c--text-only and c--icon-only modifiers to display and hide content.

Commit:
Simplify class names

Commit:
Add support for multiple radial progress bars

Commit:
Fix Firefox and Safari issues

Commit:
Merge branch 'radial-refactor' into create-progress-bar

# Conflicts:
#	progress-bar__spinner.dust
#	tests/visual/tests.dust

Commit:
Fix animation throttling issue

Commit:
Add RTL support

Commit:
Add configurable variables to readme

Commit:
Add c--icon-only and c--text-only modifiers

Commit:
Grab all options from element

This way, the user only needs to worry about providing options in one place instead of having it split between js and html.

Commit:
Update readme to reflect API changes

Commit:
Validate state in setState

Commit:
Renamed spinner -> radial progress bar

Commit:
Add the ability to set progress to the RTL tests

Commit:
Clear interval after the progress bar is full

ry5n commented:
Firstly @cdegit I want to say this is seriously impressive work. Well done! You’ve totally got the idea of this as a component with an API, public methods, events, styling hooks etc. You’ve considered accessibility and even RTL. Excellent. I have some specific comments, sorry to be rather a late reviewer.

ry5n commented on README.md:
````diff
@@ -1 +1,134 @@
+) {
+    ProgressBar.init($el);
+});
+```
````
This won’t be necessary in the general case, because Adaptive takes care of invoking the UI scripts for any components it finds. That’s what the `data-adaptivejs-component` attribute is for. We wanted our components to work automatically (UI scripts can be invoked manually in order to pass complex options if needed by the component’s API).

ry5n commented on _progress-bar.scss:
```diff
@@ -0,0 +1,252 @@
+// ---
+
+// Override Font Awesome's default path
+$fa-font-path: '/bower_components/font-awesome/fonts';
```
Using FontAwesome here raises a really interesting question around icons. Many components will need icons, and we don’t have a compelling story yet around what Stencil should do for its icon assets. For the moment, I’ll say I’m not crazy about using FontAwesome (or any particular icon set) as a requirement for using a Stencil component, but as noted, we don’t have a preferred approach worked out.

Don’t change this yet, but I’d like to revisit it after the Icon component is done and before releasing this officially.

jeffkamo replied:
Not to mention that if you decide not to use font-awesome, then you're stuck with `@import font-awesome/...` still running when you don't need it.

jeffkamo replied:
Can we put this and the `@import` into an if statement? That we we can configure the path to `null` or `false` and then not import the font awesome styles.

ry5n commented on _progress-bar.scss:
```diff
@@ -0,0 +1,252 @@
+        margin-right: 10px;
+    }
+
+    .c-progress-bar__radial-background {
```
We strongly prefer stand-alone class selectors, even for sub-components. The exception is for sub-components of a modified component, e.g. `.c-card.c--fancy .c-card__title {…}`. Normally `.c-progress-bar__radial-background` should stand alone, without nesting – this follows a general principle of keeping selector specificity low to avoid getting snarled up with the cascade.

ry5n commented on _progress-bar.scss:
```diff
@@ -0,0 +1,252 @@
+        transform: translate(-50%, -50%);
+    }
+
+    .c--progress & {
```
`.c--progress` is a modifier, and should not be used as a selector on its own. Our convention is to always chain modifiers, so this should be (I think) `.c-progress-bar.c--progress &`.

ry5n commented on _progress-bar.scss:
```diff
@@ -0,0 +1,252 @@
+//
+// 1. When the radial progress bar is shown, we want to hide the bar but keep the text
+
+.c-progress-bar__linear {
```
I find it a little bit counter-intuitive to find the linear element present in the radial variation. I wonder if the way the two variations are implemented could be made more intuitive for styling purposes.

cdegit replied:
Good point. I've created a new element that holds the label and the linear progress bar so the `.c-progress-bar__linear` element is no longer included in the radial variation.

ry5n commented on _progress-bar.scss:
```diff
@@ -0,0 +1,252 @@
+    flex-grow: 1;
+    margin-right: 5px;
+
+    .c-progress-bar__background {
```
Again, these selectors should not be nested since they don’t pertain to modifiers.

ry5n commented on bower.json:
```diff
@@ -0,0 +1,41 @@
+    "level": ">=0.1.0",
+    "spline": "~1.0.1",
+    "font-awesome": "~4.3.0",
+    "zeptojs": "1.1.6",
```
This raises a question that we need to address for Stencil as a whole: what selector library they should depend on. Adaptive 2.0+ supports using both Zepto and jQuery, with jQuery as the default. I think we should use jQuery for the component’s own tests, and it should work with both jQuery. Need to double-check, but I believe with Zepto Adaptive uses a custom build of Zepto with at least the data module built in. Certainly I think we should avoid components depending on custom builds of Zepto.

contextlimit replied:
@ry5n +1

ry5n commented on progress-bar-ui.js:
```diff
@@ -0,0 +1,337 @@
+    };
+
+    var _uuid = (function() {
+        var counter = 0;
```
I’m not the best JS dev in the world – does this counter increment safely for all instances, ensuring each instance gets unique IDs?

contextlimit replied:
That it does!

ry5n commented on progress-bar-ui.js:
```diff
@@ -0,0 +1,337 @@
+
+            // If not already initialized, create it and expose it through the data method.
+            // Also, expose a separate instance for each progress bar
+            $el.each(function(index, progressBar) {
```
As noted, Adaptive will call init for each component instance, so we shouldn’t need to iterate over anything here. AFAIK, we should be able to stick to the pattern outlined in the [Select component](https://github.com/mobify/stencil-select/blob/dev/select-ui.js#L40).

contextlimit replied:
Yah, seems to work fine to me.

ry5n commented on progress-bar.dust:
```diff
@@ -0,0 +1,19 @@
+<div {?id}id="{.}"{/id} class="c-progress-bar c--{state} {?isRadial}c--radial{/isRadial} {class}" role="progressbar" aria-valuenow="{value}" aria-valuemin="0" aria-valuemax="1" aria-valuetext="{label}" data-adaptivejs-component="stencil-progress-bar">
```
I’d like to start standardizing the order of common attributes, with `class` followed by `id`, then others. This is minor, but it will help to keep things easy to scan.

cdegit replied:
Okay, that makes sense to me. I believe the attributes were in this order when I generated the component, so we'll need to make sure to change this in the generator.

jeffkamo replied:
I've created a PR in the generator to do so.

ry5n commented on README.md:
````diff
@@ -1 +1,134 @@
+`$el` is the progress bar element. The component's instance is accessible through this DOM node. For instance, the following calls the `setProgress` method on an instance of the Progress Bar component:
+
+```js
+$('.c-progress-bar').data('progressbar').setProgress(0.5);
````
The name of the data key on which to access the UI script instance hasn’t been specced, but it should be. For the components I’ve been building I’ve been just using `component` so you’d write `$('.c-progress-bar').data('component')` to get it, and it would always be the same key for all components (can’t have two components on the same node). The alternative is what you have here: use the component name, in which case I think it should match the name exactly (minus the component namespace): `progress-bar`.

This is probably a wider discussion, but I have a feeling you will have an opinion on it :)

contextlimit replied:
> can’t have two components on the same node

It seems common enough to initialize things like jQuery plugins on the same node. Otherwise you may have to go in and add unnecessary wrappers, if possible without finding a way to hack it in there if you're not in control of the nodes. I feel like `$('.c-progress-bar').data('progress-bar')` makes sense. It also allows you to catch errors more easily in the case you might switch the component on the node, but don't notice.

ry5n replied:
@e Can you describe a use case for initializing more than one plugin on the same node?

The way I think of components is to imagine that they are custom HTML elements. It’s not possible to have say a `<select>` be something else at the same time. That’s the paradigm the framework has been working towards.

contextlimit replied:
Hmm. What about something to enforce that? Like a check when setting `.data('component', x)`? Maybe even checking `.parents()`, if it's a component where "embedding" isn't supported. I can see the case that some of these components are granular enough that they could, and would be, combined without knowing it's not intended. Maybe something is a progress bar but you also want it to be a button or badge. Maybe something is an alert but also draggable (assuming that's a component).

ry5n replied:
Right now, the way components are architected, the recommended way to use them is via their dust helper. So a button component is added to the page with this Dust: `{@c-button}My button label{/c-button}`. If the button were to ship with a UI script, it would automatically be invoked by Adaptive against that DOM node. You could invoke it yourself, if you wanted to pass custom options, like this:

``` javascript
var $progress = $('#my-progress-bar');
var customOptions = {/* some options */};
window.Adaptive.Component.init($progress, customOptions);
```

So the framework explicitly discourages invoking the component’s init function directly, and mixing and matching the behaviour of one component with the DOM of another. The [boilerplate init implementation I have in the Select component](https://github.com/mobify/stencil-select/blob/dev/select-ui.js#L36) (which I’d like to evolve into the standard) presently looks like this (so you can’t double-instantiate):

```
return {
    'init': function($el, options) {
        // If already initialized, return the instance; otherwise, create it
        // and expose it through `$('.c-select').data('component')`.
        return $el.data('component') || $el.data('component', new Select($el, options));
    }
};
```

contextlimit replied:
@ry5n You're completely right. That makes sense.

ry5n commented:
This is really really complete for an initial implementation. It’s exciting!

Here are the highlights of my review:
- Look into clarifying the relationship between the radial and linear elements; this could be suited to a short brainstorm meeting
- Need to qualify modifiers (`.c-component.c--modifier`) and avoid nesting sub-components
- Can assume the init method is invoked per-instance
- Need to decide one the data key on which the component instance is exposed
- I’d like to get an engineer to review the UI script, especially for the API.
- Design-wise, I think when the radial indicator reaches 100% it gets really cramped by the surrounding ring – it’s also very tight in the design, so this may be a case where we need to tweak the design, either by reducing the font size or increasing the size of the radial element.
- tweaks to readme

I think this can be ready for release soon.

contextlimit committed:
Minor convention improvements

contextlimit committed:
Minor instruction change for UI scripts in readme

contextlimit commented on progress-bar-ui.js:
```diff
@@ -0,0 +1,336 @@
+            if ($.browser.firefox) {
+                var $animation = $el.find('animateTransform');
+                $animation.attr('fill', 'remove');
+                $animation.attr('onend', '$(this).trigger("progressend")');
```
How do we know that $ here will be ours, not desktop $? Can we add a comment with docs on how 'fill' and 'onend' work?

cdegit replied:
I believe $ will always be ours as we are using require.js to define it.

Commit:
Add comment with links to fill, onend docs

Commit:
Move id after class

Commit:
Refactor SCSS - move subcomponents out

Commit:
Change linear progress bar to use scaleX(-1) for RTL

Commit:
Don't iterate over elements in ui.init

Commit:
Change data('progressbar') to data('component')

Commit:
Separate radial and linear progress bars

Commit:
Use jQuery for tests instead of Zepto

ry5n commented:
@cdegit Just a note on the attribute order (which you’ve adjusted) – yup, the generator needs updating. Needs updating in a few other places as well (I opened [an issue](https://github.com/mobify/generator-adaptivejs/issues/287) so we don’t forget)

jeffkamo commented:
@cdegit well click on the retry button in the "Clicking on the retry button triggers the progress-retry event" test, should there be an animation or something to indicate that the event triggered? At the moment, nothing happens.

jeffkamo commented on _progress-bar.scss:
```diff
@@ -0,0 +1,279 @@
+    display: flex;
+    align-items: center;
+    height: $tap-size;
+    margin: 0 15px;
```
I'm not sure I like this margin. Might be a bit presumptuous to add spacers like this; I could foresee a design that wants  this component to be flush with the left most side of its container.

![screen shot 2015-08-07 at 2 42 10 pm](https://cloud.githubusercontent.com/assets/734535/9146549/855dfbe0-3d12-11e5-9f86-9f13d0c29825.png)

jeffkamo replied:
Compared to the mockup:

![screen shot 2015-08-07 at 3 04 34 pm](https://cloud.githubusercontent.com/assets/734535/9146910/b087780c-3d15-11e5-9327-abee9ada352b.png)

jeffkamo commented on _progress-bar.scss:
```diff
@@ -0,0 +1,279 @@
+
+        width: 45px;
+        height: 45px;
+        margin-right: 10px;
```
would be nice to see a comment that explains the reason why properties like this are needed. In isolation, it looks a bit arbitrary.

jeffkamo commented on _progress-bar.scss:
```diff
@@ -0,0 +1,279 @@
+        position: relative;
+
+        width: 45px;
+        height: 45px;
```
These values should be replaced with their variable equivalent, in this case `$tap-target`.

Same goes for any other instances of `10px`, `15px`, or `5px`, which should probably be based on `$unit`

jeffkamo commented on _progress-bar.scss:
```diff
@@ -0,0 +1,279 @@
+// ---
+
+.c-progress-bar__linear-background {
+    position: relative;
```
I generally recommend that a comment be added to explain the intended purpose for `position` properties.

It's not always obvious why these are here, or which elements/classes they are meant to interact with. Something like... `Position relative, so the child .c-progress-bar__linear-progress element can be absolutely positioned` and or any other reasons.

jeffkamo commented on _progress-bar.scss:
```diff
@@ -0,0 +1,279 @@
+    }
+
+    .c-progress-bar__extra & {
+        position: absolute;
```
This feels dangerous to me since the text will no longer be able to influence the size of its container.

Can we try and achieve bottom-aligned text another way? We could probably use `align-self: flex-end;` on the `.c-progress-bar__extra` instead.

jeffkamo commented on _progress-bar.scss:
```diff
@@ -0,0 +1,279 @@
+
+.c-progress-bar__text {
+    .c-progress-bar__radial & {
+        position: absolute;
```
This on the other hand is fine, since the size of the text is irrelevant in regards to how this variation behaves. So no problems here.

jeffkamo commented on _progress-bar.scss:
```diff
@@ -0,0 +1,279 @@
+// ---
+//
+// This section is displayed beside the horizontal bar and is used to show the progress indicator text or icon
+// 1. When the radial progress bar is shown, we want to hide this section
```
This line comment isn't being used. I suppose it's meant to describe the below `display: none`?

jeffkamo replied:
Also, I think you need an extra blank line above these ordered lists, otherwise Github interprets the two lines as a single paragraph.

jeffkamo commented on _progress-bar.scss:
```diff
@@ -0,0 +1,279 @@
+    position: relative;
+
+    width: 35px;
+    height: 25px;
```
What is this height for?

jeffkamo commented on _progress-bar.scss:
```diff
@@ -0,0 +1,279 @@
+    width: 25px;
+    height: 25px;
+
+    text-align: center;
```
Disabling the above `width`, `height` and `text-align` seems to not do anything. What are these intended to do?

jeffkamo commented on _progress-bar.scss:
```diff
@@ -0,0 +1,279 @@
+    display: none;
+
+    width: 25px;
+    height: 25px;
```
@ry5n any concern that this is too small of a tap target for the "refresh" button?

jeffkamo commented on progress-bar.dust:
```diff
@@ -0,0 +1,24 @@
+        {@c-progress-bar__radial/}
+    {/isRadial}
+
+    <div class="c-progress-bar__middle">
```
I wonder if `middle` might be a confusing sub-class name? It's not always the middle, and the other elements at this nesting level aren't following any kind of ordering naming convention. Visually, it always looks like it's on the left or right.

jeffkamo commented on progress-bar.dust:
```diff
@@ -0,0 +1,24 @@
+
+        {^isRadial}
+        <div class="c-progress-bar__linear">
+            <div class="c-progress-bar__linear-background">
```
Is this container necessary? Couldn't you just apply the background and positioning properties on the `.c-progress-bar__linear` container?

Commit:
Fix retry event

cdegit committed:
Make radial progress bar size relative to $unit

cdegit committed:
Remove margin on c-progress-bar

cdegit committed:
Remove c-progress-bar__middle

cdegit committed:
Add a comment explaining positioning

cdegit committed:
Allow text in extra section to influence size

cdegit committed:
Properly scope svg styles

cdegit commented:
Hi @jeffkamo! I finally got the chance to review your comments and update this. I think I've changed everything you mentioned. When you get the chance, please let me know what you think. :)

cdegit committed:
Fix radial progress bar RTL margin

contextlimit commented:
:shipit:

jeffkamo commented:
Sweet, thanks for getting back to this @cdegit :)

Does font awesome still need to be replaced with the icon component?

cdegit commented:
No problem @jeffkamo. :) Yup, font awesome still needs to be replaced.

jeffkamo commented:
@cdegit few more comments above ^ (sorry, I accidentally commented in the commit instead of in the PR)

cdegit committed:
Remove $unit

cdegit committed:
Prefix circle variables

cdegit committed:
Include circle variables in configurable variables

cdegit commented:
@jeffkamo Okay, I've updated the variables. 
Is it a good idea to have variables that rely on $unit placed before $unit is defined? It all still works, but I wonder if it's a little confusing.

jeffkamo commented:
@cdegit Ah, of course, you have a good point. Maybe it just makes more sense to have all the local variables declared after the imports. That may be something we should also fix in [the current standard](https://github.com/mobify/generator-adaptivejs/blob/components/component/templates/_component.scss) too. Thoughts @ry5n?

ry5n commented:
I thought there was some reason why we needed the variables declared before the imports. I’m not sure what the reason was though. Certainly it makes more sense to me to have the dependencies first.

jeffkamo commented:
Maybe that was to override default variables? In this case, I don't believe we are overriding any default values.

ry5n commented:
Yeah, we should only be using default variables, or defining new component-local vars.

cdegit committed:
Move configurable variables after dependencies
